### PR TITLE
New regression tests for V2 surface coldstart files

### LINKED ISSUE #1977: New regression tests for v2 surface coldstart files
## Description
To support fractional grids, a new coldstart surface file format has been created. 

## Solution
Three new regression tests are proposed. One based on the `control_p8` test, one based on the `control_c48` test, and one based on `cpld_control_p8`. The coldstart files for the new tests will be based on the same test case (i.e., same input data to chgres) as the current tests. 

## Related to
The new coldstart file format requires an update to an FV3 io routine. See:
- NOAA-EMC/fv3atm#670
- NOAA-EMC/fv3atm#698

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## PR Author Checklist:

- [x] I have linked PR's from all sub-components involved in section below. 
- [x] I am confirming reviews are completed in ALL sub-component PR's.
- [x] I have run the full RT suite on either Hera/Cheyenne AND have attached the log to this PR below this line:
  - LOG: Baselined to my branch. [RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/14173163/RegressionTests_hera.log)
- [x] I have added the list of all failed regression tests to "Anticipated changes" section.
- [x] I have filled out all sections of the template.

## Description
The UFS_UTILS chgres_cube program was updated for fractional (partial land/non-land) grids and a new version of the surface coldstart file is output. This new file format will work with either fractional or non-fractional cases. Create three new regression tests that use the new file format.

## Commit Message
```
- Add three new regression tests for version 2 of the surface coldstart file. (Closes #1977)
```

### Associated UFSWM Issue to close
- Fixes #1977.

### Subcomponent Pull Requests
- NOAA-EMC/fv3atm#698

### Blocking Dependencies
N/A

### Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Input data
- [ ] No changes are expected to input data.
- [x] Changes are expected to input data:
  - [x] New input data.
  - [ ] Updated input data.

### Regression Tests:
- [x] No changes are expected to any current regression test.
- [ ] Changes are expected to the following tests:

Tests effected by changes in this PR:</summary>
- Three new regression tests based on the existing 'control_c48', 'control_p8' and 'cpld_control_p8' tests

New coldstart files for these tests are in `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/RT`
- control_c48.v2.sfc - `./FV3_input_data48/INPUT_L127.v2.sfc`
- control_p8.v2.sfc - `./FV3_input_data/INPUT_L127.v2.sfc`
- cpld_control_p8.v2.sfc - `./FV3_input_data/INPUT_L127_mx100.v2.sfc`

New baseline files will need to be hosted for these new tests.

All regression tests passed except for the three new tests (because the new tests do not have baseline files yet).

### Libraries

- [x] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

<details><summary>Code Managers Log</summary>

- [ ] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.
  - [ ] N/A

### Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [ ] Gaea
  - [ ] Cheyenne
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment
</details>

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@GeorgeGayno-NOAA We're adding a Commit Message requirement to the PRs. Please add one in the space provided.

**Comment by GeorgeGayno-NOAA** [comment]
Ready for review.

**Comment by GeorgeGayno-NOAA** [comment]
@jkbk2004 - I was told your group handles PRs. I am unable to assign this PR to you or anyone else.

File: tests/fv3_conf/control_run.IN [hunk 1]
```diff
@@ -13,10 +13,16 @@ fi
 echo "inputdir=$inputdir,NPX=$NPX"
 
 OPNREQ_TEST=${OPNREQ_TEST:-false}
+V2_SFC_FILE=${V2_SFC_FILE:-false}
 SUFFIX=${RT_SUFFIX}
 if [ $WARM_START = .false. ]; then
   mkdir INPUT RESTART
-  cp -r @[INPUTDATA_ROOT]/${inputdir}/INPUT_L127/* ./INPUT/.
+  if [ "$V2_SFC_FILE" = "true" ]; then
```
Comment (GeorgeGayno-NOAA) [inline tests/fv3_conf/control_run.IN:20]: The 'coldstart' data for the new tests currently resides in a personal directory. These files will need to be placed in the official directory. Then, I can update this file.

Comment (BrianCurtis-NOAA) [reply tests/fv3_conf/control_run.IN:20]: Are these data replacing/overwriting older ones, or brand new?

Comment (BrianCurtis-NOAA) [reply tests/fv3_conf/control_run.IN:20]: I see your comments in the PR, nevermind.

**Comment by GeorgeGayno-NOAA** [comment]
I can't get the regression tests to complete. 

All tests appear to run (see log files in `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/global-workflow.git/sorc/ufs_model.fd/tests/logs/log_hera`). 

However, the RegressionTests_hera.log file is never updated. 

I piped the output from the rt.sh command to `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/global-workflow.git/sorc/ufs_model.fd/tests/output` but I can't figure out what went wrong.

**Comment by jkbk2004** [comment]
@GeorgeGayno-NOAA I was able to run rt.sh on hera for this pr: /scratch2/NCEPDEV/marine/Jong.Kim/UFS-RT/rt-2005/tests/logs/RegressionTests_hera.log. Is the issue on workflow side ?

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA I was able to run rt.sh on hera for this pr: /scratch2/NCEPDEV/marine/Jong.Kim/UFS-RT/rt-2005/tests/logs/RegressionTests_hera.log. Is the issue on workflow side ?

I can run individual tests. But when I try to run the full suite, the tests seem to complete, but something fails before the RegressionTests_hera.log is created. That log should contain the results of all 240+ tests, but it does not.

**Comment by GeorgeGayno-NOAA** [comment]
> I can't get the regression tests to complete.
> 
> All tests appear to run (see log files in `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/global-workflow.git/sorc/ufs_model.fd/tests/logs/log_hera`).
> 
> However, the RegressionTests_hera.log file is never updated.
> 
> I piped the output from the rt.sh command to `/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/global-workflow.git/sorc/ufs_model.fd/tests/output` but I can't figure out what went wrong.

The problem starts just after some rocoto warnings:

```
+ /apps/rocoto/1.3.6/bin/rocotorun -v 10 -w /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/global-workflow.git/sorc/ufs_model.fd/tests/rocoto_workflow.xml -d /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/global-workflow.git/sorc/ufs_model.fd/tests/rocoto_workflow.db
01/31/24 19:23:31 UTC :: rocoto_workflow.xml :: WARNING: <partition> has empty content and is ignored
01/31/24 19:23:31 UTC :: rocoto_workflow.xml :: Submitting compile_s2swa_32bit_intel using sbatch < /tmp/sbatch.in20240131-233930-170suuj with input
{{

```

**Comment by DusanJovic-NOAA** [comment]
Can you try to run regression test with ecflow:

./rt.sh -e ....

File: tests/fv3_conf/cpld_control_run.IN [hunk 2]
```diff
@@ -25,7 +27,11 @@ else
      MOM_IC=@[INPUTDATA_ROOT]/MOM6_IC/${OCNRES}/${SYEAR}${SMONTH}${SDAY}${SHOUR}
      ICE_IC=@[INPUTDATA_ROOT]/CICE_IC/${OCNRES}/${SYEAR}${SMONTH}${SDAY}${SHOUR}
   else
-     FV3_IC=@[INPUTDATA_ROOT]/${FV3_DIR}/INPUT_L127_mx${OCNRES}
+     if [ "$V2_SFC_FILE" = "true" ]; then
+       FV3_IC=/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/RT/FV3_input_data/INPUT_L127_mx100.v2.sfc
```
Comment (DeniseWorthen) [inline tests/fv3_conf/cpld_control_run.IN]: These paths will need to be reverted once the data is in place and we are ready to start testing this PR.

Comment (jkbk2004) [reply tests/fv3_conf/cpld_control_run.IN]: Can we agree and reset directory naming convention to use underscore (not dot) like INPUT_L127_mx100_v2_sfc?

**Comment by GeorgeGayno-NOAA** [comment]
> Can you try to run regression test with ecflow:
> 
> ./rt.sh -e ....

That worked. Thanks.

File: tests/fv3_conf/cpld_control_run.IN (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: If you can place the data for the new tests in the official location, I can update my branch to point to it.

Comment (jkbk2004) [reply tests/fv3_conf/cpld_control_run.IN]: I renamed the new input directories. Can you adjust like FV3_IC=@[INPUTDATA_ROOT]/${FV3_DIR}/INPUT_L127_mx${OCNRES}_v2_sfc ?

File: tests/fv3_conf/control_run.IN [hunk 3]
```diff
@@ -13,10 +13,16 @@ fi
...
+    cp -r /scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/chgres_cube.fractional/RT/${inputdir}/INPUT_L127.v2.sfc/* ./INPUT/.
+#   cp -r @[INPUTDATA_ROOT]/${inputdir}/INPUT_L127.v2.sfc/* ./INPUT/.
```
Comment (jkbk2004) [inline tests/fv3_conf/control_run.IN]: You can re-activate this line with cp -r @[INPUTDATA_ROOT]/${inputdir}/INPUT_L127_v2_sfc/* ./INPUT/.

File: tests/fv3_conf/cpld_control_run.IN (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: Done at 9f9f8f2.

Comment (GeorgeGayno-NOAA) [reply tests/fv3_conf/cpld_control_run.IN]: And de655ec.

**Comment by jkbk2004** [comment]
@GeorgeGayno-NOAA can you sync up branch? you can keep your RegressionTests_hera.log.

**Comment by jkbk2004** [comment]
@GeorgeGayno-NOAA Note that feature branch is recommended but your branch syncs ok. Then, we can move on to commit. We will try to combine in #2115 to this pr after your branch is synced up. @BrianCurtis-NOAA FYI: #2111 is combined ok into #2115

File: tests/tests/control_p8.v2.sfc [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+export MOM6_RESTART_SETTING=n
+# P8 (not used for standalone)
+export USE_CICE_ALB=.false.
+
+export WRITE_NSFLIP=.true.
+
+export FV3_RUN=control_run.IN
+export CCPP_SUITE=FV3_GFS_v17_p8
+export FIELD_TABLE=field_table_thompson_noaero_tke
+export DIAG_TABLE=diag_table_p8_template
```
Comment (jkbk2004) [inline tests/tests/control_p8.v2.sfc]: There is no diag_table_p8_template. diag_table_cpld.IN can be used.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA Note that feature branch is recommended but your branch syncs ok. Then, we can move on to commit. We will try to combine in #2115 to this pr after your branch is synced up. @BrianCurtis-NOAA FYI: #2111 is combined ok into #2115

My branch looks synced up. Did you do that?

**Comment by jkbk2004** [comment]
> > @GeorgeGayno-NOAA Note that feature branch is recommended but your branch syncs ok. Then, we can move on to commit. We will try to combine in #2115 to this pr after your branch is synced up. @BrianCurtis-NOAA FYI: #2111 is combined ok into #2115
> 
> My branch looks synced up. Did you do that?

@GeorgeGayno-NOAA I pushed a few commits to sync up. This pr is ready. Hera full test passes ok. There was a minor issue on #2115 side. The issue was fixed but we are not going to combine with any other PRs. @BrianCurtis-NOAA FYI @zach1221 @FernandoAndrade-NOAA  can you pick up from here.

File: tests/tests/control_p8.v2.sfc (hunk 4 above)
Comment (jkbk2004) [reply tests/tests/control_p8.v2.sfc]: fixed

File: tests/fv3_conf/control_run.IN (hunk 3 above)
Comment (jkbk2004) [reply tests/fv3_conf/control_run.IN]: fixed

**Comment by zach1221** [comment]
Hey, @BrianCurtis-NOAA . How are acorn/wcoss coming along?

**Comment by BrianCurtis-NOAA** [comment]
gnv1_nested is causing issues on WCOSS2. I've tried 2 comparisons, redid that baseline and compared against it again and it still failed in comparison. This test needs to be disabled on WCOSS2 and the developer of that test needs to find out why it isn't reproducing it's own baselines.

**Comment by zach1221** [comment]
> gnv1_nested is causing issues on WCOSS2. I've tried 2 comparisons, redid that baseline and compared against it again and it still failed in comparison. This test needs to be disabled on WCOSS2 and the developer of that test needs to find out why it isn't reproducing it's own baselines.

Brian, I can make an issue and make note of each machine this is occurring on.

**Comment by zach1221** [comment]
The Hera ORTs should be done shortly and we can move on to start merging.

**Comment by zach1221** [comment]
Testing is complete. Moving to merge the fv3atm sub-pr.

**Comment by zach1221** [comment]
@GeorgeGayno-NOAA fv3atm PR #698 has been merged. Please update the submodule hash and revert the .gitmodule url.
https://github.com/NOAA-EMC/fv3atm/commit/b8f7b0f33df6dc712f241dcd6818c9ec27134e26

**Review by jkbk2004 — APPROVED** [review]

**Review by zach1221 — APPROVED** [review]

**Pull request merged**